### DISCUSSION TITLE
Wifi Devices: Share Your Experience

### DISCUSSION THREAD

**Discussion by shanemcw** [shanemcw/homebridge-amazondash-mac#1] [General]
Have a wifi device work or not work?

No drivers for Linux? Does not support monitor mode? Is yours working great? Did you find a good deal?

Share your wifi devices experiences here.

| Works  | Model | Price | Tested  | Notes |
| :---:  | :---  | ---:  | ---:  |  :---  |
| ✅ | Panda Wireless N PAUO5 | $18  | 2021  | Worked with Ubuntu out-of-the-box. |
| ✅ | Panda Wireless N PAUO6 | $19  | 2021  | Similar to PAU05 with an external antenna. As with PAU05, worked with Ubuntu out-of-the-box.  |
| 🤔 |  TP-Link TL-WN725N | $10 |  | Not yet tested |
| 💀 | Netis WF2123 N300 | $15  | 2021  | No Linux drivers nor support.  |
| ❔ |  |  |  |  |
| ❔ |  |  |  |  |
| ❔ |  |  |  |  |

* ✅ Recommended
* 👎 Cumbersome to get working
* 💀 Could not get working
* 🤔 Looks interesting… anyone want to try it?
* ❔Your next review

**Comment by TLam-projects** [comment]
@shanemcw little bit of a pain getting it to work on Pi 3 but this guide was decent to get it up and running: https://null-byte.wonderhowto.com/how-to/enable-monitor-mode-packet-injection-raspberry-pi-0189378/

The only annoying thing is that tcpdump will suddenly stop after a while (Probably about a week or so?). Forces me to restart the Pi.

Haven't found a solution to it yet..

**Reply by shanemcw** [reply]
Thanks tom288 for sharing your insight. I'll roll a solution into the next release, i.e. re-spawning a new tcpdump process if the prior one exits. Please take a look at the logs and let me know if you see the "exit" and "close" messages at the point when tcpdump stops running. Any out-of-the-ordinary error messages produced by tcpdump are also injected into the logs—pass those along if you see any.

Before you restart the Pi next time, try restarting the Homebridge server via the Homebridge UI dashboard—that can be done remotely and will spawn a new tcpdump process.

Please take notes and I'll include your insights in the plugin's readme for other Pi users!

**Reply by TLam-projects** [reply]
So it's a little bit weirder than that..
Installing re4son-kernel grants the ability to create an interface on the Pi using their script mon0up. I can then test that monitor mode on the interface is working using airodump-ng mon0.

Then using mon0 as the interface for monitoring, I can get tcpdump to monitor using that interface.

After a while, it eventually stops. Logging back into the pi and testing airodump and tcpdump seems to give me nothing until I restart the pi. I did think that the logs weren't rotating and getting too big.. But your JS seems to rotate them. The only crude method I have at the moment is restarting the pi each time it stops.

**Reply by shanemcw** [reply]
Thanks for the update!
Note: You shouldn't need airodump-ng to test—testing the successful monitoring of the Dash Button activity using the wifi device in monitor mode by tcpdump on the command line by itself works too.
If you are seeing that tcpdump runs in command line (remember the --monitor-mode), but there are no results at all (or an error?), that looks like the wifi device is no longer working correctly (that maybe somehow restored by a reboot). This of course is different than the tcpdump process stopping (i.e. you no longer see it as a running process using ps). 
The dashboard of the Homebridge UI shows how much CPU and memory is being used—keep an eye on that.
To simplify the setup I am using, I use a $10 USB wifi dongle—if it is flakey I can try a different one (and I get to use the native wifi device to connect to the network).

**Reply by shanemcw** [reply]
I see some chatter in this forum https://www.raspberrypi.org/forums/viewtopic.php?t=133103 that monitor mode support on the Pi is a bit flakey due to driver differences between Linux and Raspberry. The forum ends with a post recommending the USB Panda Wireless N PAUO5—which is coincidentally the device I am using with my Ubuntu setup.
